tools/functional-tester/etcd-agent/agent.go

```diff
@@ -75,23 +77,59 @@ func (a *Agent) start(args ...string) error {
 	return nil
 }
 
+var alreadyFinished = errors.New("os: process already finished")
+
 // stop stops the existing etcd process the agent started.
 func (a *Agent) stop() error {
 	if a.state != stateStarted {
 		return nil
 	}
-	err := a.cmd.Process.Kill()
+
+	err := waitCmdExit(a.cmd, syscall.SIGTERM)
 	if err != nil {
 		return err
 	}
-	_, err = a.cmd.Process.Wait()
+
+	err = a.cmd.Process.Kill()
+	if err.Error() == alreadyFinished.Error() {
+		a.state = stateStopped
+		err = nil
+	}
+	return err
```

Comment:
It's possible this function will return before the `Wait()` returns; the process could still be holding its tcp ports.